lib/list.h

```diff
@@ -98,6 +98,10 @@ static inline struct metal_list *metal_list_first(struct metal_list *list)
 	     (node) != (list);			\
 	     (node) = (node)->next)
 
+#define metal_list_for_each_safe(list, node, temp)		\
```

Comment:
While other APIs are not documented, I prefer we document this API in doxygen format. This will clearly say what is head, node etc..
Ideally I would prefer if we keep order of arguments matching with kernel API of list_for_each_safe. It makes developer's life easy.

https://github.com/torvalds/linux/blob/2cf0f715623872823a72e451243bbf555d10d032/include/linux/list.h#L723

Reply:
I agree with both of Tanmay's points.

Reply:
Yes make sense
Please @yintao707, update based on @tnmysh remarks

Reply:
@tnmysh @arnopo OK, we will update this PR, include two commits:
1. document the list API in doxygen format.
2. add metal_list_for_each_safe() support.
Change the arguments order will affect a lot of code, so let's change this in the other PR,

Reply:
> Yes make sense Please @yintao707, update based on @tnmysh remarks

hi, @arnopo , @tnmysh, @edmooring , 
I have updated this PR， I am not sure if the comments I added are compliant.  Please help me review it again, thanks.​
The parameters of this API need to be in the same order as the parameters of the Linux API, I think that metal_list_for_each should be modified firstly, this can ensure that the two APIs are consistent,  howerver, which will have a lot of impact. so,  let's change this in the other PR.